src/builtins/shell/README.md

```diff
@@ -0,0 +1,130 @@
+
+```typescript
+const output = Shell({command: '/usr/local/bin/sampler'});
+const result = await output.execute([
+  {
+    timestamp: '2021-01-01T00:00:00Z',
+    duration: 3600,
+    'cpu/energy': 0.002,
+    'memory/energy': 0.000005,
+  },
+]);
+```
+
+## Considerations
+
+The `shell` is designed to run arbitrary external plugins. This means IF does not necessarily know what calculations are being executed in the external plugin. There is no strict requirement on the return type, as this depends upon the calculations and the position of the external plugin in a plugin pipeline. For example, one external plugin might carry out the entire end-to-end SCI calculation, taking in usage inputs and returning `sci`. In this case, the plugin is expected to return `sci` and it would be the only plugin invoked in the `manifest`.
+
+However, it is also entirely possible to have external plugins that only deliver some small part of the overall pipeline, and rely on IF other plugins to do the rest.
+
+Since the design space for external plugins is so large, it is up to external plugin developers to ensure compatibility with IF built-ins.
+
+## Example manifest
+
+IF users will typically call the shell plugin as part of a pipeline defined in a `manifest` file. In this case, instantiating and configuring the plugin is handled by and does not have to be done explicitly by the user. The following is an example `manifest` that calls an external plugin via `shell`. It assumes the plugin takes `cpu/energy` and `memory/energy` as inputs and returns `energy`:
+
+```yaml
+name: shell-demo
+description:
+tags:
+initialize:
+  outputs:
+    - yaml
+  plugins:
+    sampler:
+      method: Shell
+      path: 'builtin'
+      global-config:
+        command: python3 /usr/local/bin/sampler
+tree:
+  children:
+    child:
+      pipeline:
+        - sampler
+      inputs:
+        - timestamp: 2023-07-06T00:00
+          duration: 1 # Secs
+          cpu/energy: 0.002
+          memory/energy: 0.000005
+```
+
+In this hypothetical example, the plugin is written in Python and invoked by executing `python3 /usr/local/bin/sampler` in a shell.
+The plugin should return an `output` looking as follows:
+
+```yaml
+name: shell-demo
+description:
+tags:
+initialize:
+  outputs:
+    - yaml
+  plugins:
+    sampler:
+      method: Shell
+      path: 'builtin'
+      global-config:
+        command: python3 /usr/local/bin/sampler
+tree:
+  children:
+    child:
+      pipeline:
+        - sampler
+      inputs:
+        - timestamp: 2023-07-06T00:00
+          duration: 1 # Secs
+          cpu/energy: 0.002
+          memory/energy: 0.000005
+      outputs:
+        - timestamp: 2023-07-06T00:00
+          duration: 1 # Secs
+          cpu/energy: 0.002
+          memory/energy: 0.000005
+          energy: 0.02 # added by plugin
+```
+
+You can run this example `manifest` by saving it as `./examples/manifests/test/sci.yml` and executing the following command from the project root:
```

Comment:
I think, the manifest path should be changed

Reply:
```suggestion
You can run this example `manifest` by saving it as `manifests/plugins/shell.yml` and executing the following command from the project root:
```